include/realizations/catchment/Bmi_Multi_Formulation.hpp

```diff
@@ -275,11 +275,30 @@ namespace realization {
          * @return The inclusive beginning of the period of time over which this instance can provide this data.
          */
         time_t get_forcing_output_time_begin(const std::string &forcing_name) override {
+            std::string var_name = forcing_name;
+            if(var_name == "*" || var_name == ""){
+                // when unspecified, assume all data is available for the same range.
```

Comment:
> when unspecified, assume all data is available for the same range.

I see a potential issue with this.  We will (eventually) implement functionality where the last module feeds something back into the first.  We will need an extra data point, at a time before the "beginning," for an item that is (at least later) a forcing output.

How we implement things later will make a difference as to whether this directly leads to a conflict, but that's incidental to the real problem.  This usage is inconsistent with the design of _ForcingProvider_, the design itself is lacking.

We need to adjust the interface to add another function to do what we are trying to force this to do, which is get a "beginning" for _the provider_.  This function should (only) get the "beginning" for a particular forcing property.

Reply:
Concur. The original implementation/first iteration attempted to implement the feature without modifying the interface. A later iteration would be better suited to modify the interface where appropriate, and the assumption of all variables having the same time range holds in the present state of the system. This PR is a bugfix on the first iteration, not a second iteration.

Reply:
I'm not super clear on the entire scope of the bugs from #311 that this is intended to address, but I consider using `get_forcing_output_time_begin` incorrectly to be a bug.   Regardless of whether it's separate from the bugs to fix in this PR, if we had sufficient test coverage in place for `Bmi_Multi_Formulation::get_forcing_output_time_begin`, I am pretty sure this change wouldn't pass.

That said, I'm alright going forward with this, as long as we are all on the same page.  I see this as not just deferring the fix of bug _A_, but actually extending bug _A_, as part of efforts to fix bug _B_.  That isn't the kind of thing we need to become desensitized to (or let ourselves do very often), so I think it's important we emphasize when it happens.

I'll plan to start working on the interface (and adding tests we are missing for `Bmi_Multi_Formulation`) once this PR is committed.